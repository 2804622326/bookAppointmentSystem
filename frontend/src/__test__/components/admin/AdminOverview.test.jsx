import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import AdminOverview from '../../../components/admin/AdminOverview.jsx';

jest.mock('../../../components/charts/RegistrationChart.jsx', () => () => <div data-testid="reg-chart" />);
jest.mock('../../../components/charts/AppointmentChart.jsx', () => () => <div data-testid="app-chart" />);
jest.mock('../../../components/charts/AccountChart.jsx', () => () => <div data-testid="acct-chart" />);
jest.mock('../../../components/charts/VetSpecializationChart.jsx', () => () => <div data-testid="spec-chart" />);

jest.mock('../../../components/user/UserService.js', () => ({
  countPatients: jest.fn(() => Promise.resolve(5)),
  countUsers: jest.fn(() => Promise.resolve(20)),
  countVeterinarians: jest.fn(() => Promise.resolve(3)),
}));

jest.mock('../../../components/appointment/AppointmentService.js', () => ({
  countAppointments: jest.fn(() => Promise.resolve(7)),
}));

test('displays fetched counts with snapshot', async () => {
  const { asFragment } = render(<AdminOverview />);
  await waitFor(() => screen.getByText('20'));
  expect(screen.getByText('20')).toBeInTheDocument();
  expect(screen.getByText('7')).toBeInTheDocument();
  expect(screen.getByText('3')).toBeInTheDocument();
  expect(screen.getByText('5')).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
