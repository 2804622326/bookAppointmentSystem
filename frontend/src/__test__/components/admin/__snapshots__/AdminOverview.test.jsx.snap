// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays fetched counts with snapshot 1`] = `
<DocumentFragment>
  <main>
    <h5
      class="chart-title"
    >
      Activities Overview
    </h5>
    <div
      class="main-cards"
    >
      <div
        class="admin-card"
      >
        <div
          class="card-inner"
        >
          USERS
          <svg
            class="card-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 14s-1 0-1-1 1-4 5-4 5 3 5 4-1 1-1 1zm4-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6m-5.784 6A2.24 2.24 0 0 1 5 13c0-1.355.68-2.75 1.936-3.72A6.3 6.3 0 0 0 5 9c-4 0-5 3-5 4s1 1 1 1zM4.5 8a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5"
            />
          </svg>
        </div>
        <h3>
          20
        </h3>
      </div>
      <div
        class="admin-card"
      >
        <div
          class="card-inner"
        >
          APPOINTMENTS
          <svg
            class="card-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 14s-1 0-1-1 1-4 5-4 5 3 5 4-1 1-1 1zm4-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6m-5.784 6A2.24 2.24 0 0 1 5 13c0-1.355.68-2.75 1.936-3.72A6.3 6.3 0 0 0 5 9c-4 0-5 3-5 4s1 1 1 1zM4.5 8a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5"
            />
          </svg>
        </div>
        <h3>
          7
        </h3>
      </div>
      <div
        class="admin-card"
      >
        <div
          class="card-inner"
        >
          VETERINARIANS
          <svg
            class="card-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 14s-1 0-1-1 1-4 5-4 5 3 5 4-1 1-1 1zm4-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6m-5.784 6A2.24 2.24 0 0 1 5 13c0-1.355.68-2.75 1.936-3.72A6.3 6.3 0 0 0 5 9c-4 0-5 3-5 4s1 1 1 1zM4.5 8a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5"
            />
          </svg>
        </div>
        <h3>
          3
        </h3>
      </div>
      <div
        class="admin-card"
      >
        <div
          class="card-inner"
        >
          PATIENTS
          <svg
            class="card-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 14s-1 0-1-1 1-4 5-4 5 3 5 4-1 1-1 1zm4-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6m-5.784 6A2.24 2.24 0 0 1 5 13c0-1.355.68-2.75 1.936-3.72A6.3 6.3 0 0 0 5 9c-4 0-5 3-5 4s1 1 1 1zM4.5 8a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5"
            />
          </svg>
        </div>
        <h3>
          5
        </h3>
      </div>
    </div>
    <div
      class="charts"
    >
      <div
        class="chart-container"
      >
        <div
          data-testid="reg-chart"
        />
      </div>
      <div
        class="chart-container"
      >
        <div
          data-testid="app-chart"
        />
      </div>
      <div
        class="chart-container"
      >
        <div
          data-testid="acct-chart"
        />
      </div>
      <div
        class="chart-container"
      >
        <div
          data-testid="spec-chart"
        />
      </div>
    </div>
  </main>
</DocumentFragment>
`;
